velox/dwio/parquet/tests/ParquetTpchTestBase.cpp

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/dwio/parquet/tests/ParquetTpchTestBase.h"
+
+namespace facebook::velox::exec::test {
+
+void ParquetTpchTestBase::SetUpTestSuite() {
+  if (duckDb_ == nullptr) {
+    duckDb_ = std::make_shared<DuckDbQueryRunner>();
+  }
+  tempDirectory_ = TempDirectoryPath::create();
+
+  functions::prestosql::registerAllScalarFunctions();
```

Comment:
I wonder if this logic has to be in static SetupTestSuite or if it can be moved back to Setup.

Reply:
@duanmeng `SetupTestSuite` I believe cause issues when running tests in parallel. Let's keep the old `SetUp`. I don't think is an issue if we create these instances per test.